### PR TITLE
UI: Ux modification

### PR DESCRIPTION
**Pull request by pratap0007**
- Remove download icon
    - Remove Home,Search and Community from Home page and unnecessary files

### CODE REVIEW

**Review by sthaha — CHANGES_REQUESTED** [review]
the UX modification includes a lot of unrelated go lang api change. could you please remove those?

**Comment by pratap0007** [comment]
> the UX modification includes a lot of unrelated go lang api change. could you please remove those?


removed unrelated go lang api change.

**Comment by pratap0007** [comment]
> the UX modification includes a lot of unrelated go lang api change. could you please remove those?

removed  unrelated go lang api change .

File: frontend/src/components/filter/Filter.tsx [hunk 1]
```diff
@@ -15,173 +14,186 @@ import {API_URL} from '../../constants';
...
-  const ddff = () => {
-    setX((x) => x + 1);
+
+  // fetch api function
+  const fetchApi = (typeurl: string, verifiedurl: string, categoryurl: any) => {
+    fetch(`${API_URL}/resources/${typeurl}/${verifiedurl}?tags=${categoryurl} `)
+        .then((resp) => resp.json())
+        .then((data) => {
+          const taskarr = data.sort((first: any, second: any) => {
+            if (first.name > second.name) {
```
Comment (sthaha) [inline frontend/src/components/filter/Filter.tsx]:
```
data.sort( (a: any, b: any) => a > b ? 1 : -1 ) 
```

should do the same in theory

File: frontend/src/components/filter/Filter.tsx [hunk 2]
```diff
@@ -15,173 +14,186 @@ import {API_URL} from '../../constants';
...
-    status.forEach((it: any) => {
-      if (it.id === e.target.id) {
-        it.isChecked = e.target.checked;
+    let categoryurl = '';
+    const target = event.target;
+    // for handling isChecked parameter of checkbox
+    setCheckBoxStatus({...checkBoxStatus, [target.value]: target.checked});
+    status.checklist.forEach((it: any) => {
+      if (it.id === event.target.id) {
+        it.isChecked = event.target.checked;
```
Comment (sthaha) [inline frontend/src/components/filter/Filter.tsx]: return soon after this line? do you expect to find more than 1 item matching the `id` ?

File: frontend/src/components/filter/Filter.tsx [hunk 3]
```diff
@@ -15,173 +14,186 @@ import {API_URL} from '../../constants';
...
-    status.forEach((it: any) => {
-      if (it.isChecked === true) {
-        it.isChecked = false;
-      }
+  //   function for clearing all checkbox
+  const clearFilter = () => {
+    setCheckBoxStatus(
+        tempObj,
+    );
+    status.checklist.map((it: any) => {
```
Comment (sthaha) [inline frontend/src/components/filter/Filter.tsx]: what is the point of mapping if the return value is not used?

what is the intend here?

File: frontend/src/components/filter/Filter.tsx [hunk 4]
```diff
@@ -15,173 +14,186 @@ import {API_URL} from '../../constants';
...
+      it.isChecked = false;
+      return status;
     });
-    setStatus(status);
-    ddff();
```
Comment (sthaha) [inline frontend/src/components/filter/Filter.tsx]: :D we had `ddff` 😂 ?

File: frontend/src/components/filter/Filter.tsx [hunk 5]
```diff
@@ -217,31 +229,21 @@ const Filter: React.FC = (props: any) => {
...
-      setShow('Show less');
-    } else {
-      setMax((x) => x - 13);
-      setShow('Show all');
-    }
-  };
 
   return (
-    <div className="filter-size" >
+    <div className="filter-size" key="">
```
Comment (sthaha) [inline frontend/src/components/filter/Filter.tsx]: why is the key empty? wouldn't this create issues? doesn't react expect keys to be unique?

File: frontend/src/components/filter/Filter.tsx (hunk 2 above)
Comment (pratap0007) [reply frontend/src/components/filter/Filter.tsx]: no,expect 1 item matching the id.

File: frontend/src/components/filter/Filter.tsx (hunk 3 above)
Comment (pratap0007) [reply frontend/src/components/filter/Filter.tsx]: intend is to iterate over array ,should be foreach

File: frontend/src/components/filter/Filter.tsx (hunk 5 above)
Comment (pratap0007) [reply frontend/src/components/filter/Filter.tsx]: here by mistake I have used and need to remove it. react expect keys to be unique.

File: frontend/src/components/filter/Filter.tsx [hunk 6]
```diff
@@ -55,13 +55,7 @@ const Filter: React.FC = (props: any) => {
...
         .then((resp) => resp.json())
         .then((data) => {
-          const taskarr = data.sort((first: any, second: any) => {
-            if (first.name > second.name) {
-              return 1;
-            } else {
-              return -1;
-            }
-          });
+          const taskarr = data.sort((a: any, b: any) => a > b ? 1 : -1);
```
Comment (PuneetPunamiya) [inline frontend/src/components/filter/Filter.tsx]: Here it should be a.name > b.name I guess as we need to sort this on the basis of name

**Pull request merged**